components/ImageUpload.tsx

```diff
@@ -1,43 +1,60 @@
 import * as ImagePicker from "expo-image-picker";
-import { Constants } from "expo-constants";
 import React, { useEffect, useState } from "react";
-import { Platform, Image } from "react-native";
+import { Platform, Image, Button, Alert } from "react-native";
 import { View, Text } from "./Themed";
 import { MaterialIcons } from "@expo/vector-icons";
 import { TouchableOpacity } from "react-native-gesture-handler";
-
+import { useIsFocused } from "@react-navigation/native";
 interface Props {
   handleUpload: (a: any) => void;
   setChosenImg?: () => void;
   changeProfilePicIsTrue?: boolean;
 }
 
-const ImageUpload = ({
+export const ImageUpload = ({
   handleUpload,
-  setChosenImg,
   changeProfilePicIsTrue,
 }: Props) => {
   const [image, setImage] = useState<any>(null);
+  const isFocused = useIsFocused();
 
   useEffect(() => {
-    PermissionImageUpload();
-  }, []);
+    isFocused ? PermissionImageUpload() : null;
+  }, [isFocused]);
 
   const PermissionImageUpload = async () => {
     try {
       if (Platform.OS !== "web") {
         const { status } =
           await ImagePicker.requestMediaLibraryPermissionsAsync();
         if (status !== "granted") {
-          alert("Permission denied");
+          alert("Åtkomst nekad");
         }
+        pickImage();
```

Comment:
This makes photo gallery open straight away. This method should only be called when user press button